LibraryWebApp/Controllers/BookController.cs

```diff
@@ -118,14 +191,28 @@ public IActionResult Edit(BookAuthorVM obj)
                         _db.BookAuthor.Add(bookAuthor);
                     }
                 }
+                if (obj.TypesOfBookId != null)
+                {
+                    foreach (var typeId in obj.TypesOfBookId)
+                    {
+                        BookType bookType = new BookType
+                        {
+                            IdBook = obj.Book.Id,
+                            IdTypeOfBook = typeId
+                        };
+                        _db.BookType.Add(bookType);
+                    }
+                }
                 _db.SaveChanges();
                 return RedirectToAction(nameof(Index));
             }
-            BookAuthorVM model = new BookAuthorVM
+            BookVM model = new BookVM
             {
                 Authors = _db.Author,
                 BookAuthors = _db.BookAuthor.Where(x => x.IdBook == obj.Book.Id),
-                Book = _db.Book.Include(x => x.BookAuthor).ThenInclude(x => x.Author).Where(x => x.Id == obj.Book.Id).FirstOrDefault()
+                Types = _db.TypeOfBook,
+                BookTypes = _db.BookType.Where(x => x.IdBook == obj.Book.Id),
+                Book = _db.Book.Include(x => x.BookAuthor).ThenInclude(x => x.Author).Where(x => x.Id == obj.Book.Id).Include(x => x.BookType).ThenInclude(x => x.TypeOfBook).Where(x => x.Id == obj.Book.Id).FirstOrDefault()
```

Comment:
you can break those functions into multiple lines so that they would not form a long snake. Add new line before '.'